aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,271 @@
+author: pavelsavara
+description: Learn how to run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop.
+monikerRange: '>= aspnetcore-7.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/18/2022
+uid: client-side/import-export-interop
+---
+# Run .NET from JavaScript using `[JSImport]`/`[JSExport]` interop
+
+This article explains how to run .NET from JavaScript (JS) using `[JSImport]`/`[JSExport]` interop.
+
+Existing JS apps can use the expanded client-side WebAssembly support in .NET 7 to reuse .NET libraries from JS or to build novel .NET-based apps and frameworks.
+
+> [!NOTE]
+> This article focuses on running .NET from JS apps without any dependency on [Blazor](xref:blazor/index). 
+
+<!--
+
+    HOLD: ADD TO NOTE AFTER BLAZOR ARTICLE IS MERGED ...
+
+    For guidance on using `[JSImport]`/`[JSExport]` interop in Blazor WebAssembly apps, see <xref:blazor/js-interop/import-export-interop>.
+
+-->
+
+These approaches are appropriate when you only expect to run on WebAssembly (:::no-loc text="WASM":::). Libraries can make a runtime check to determine if the app is running on :::no-loc text="WASM"::: by calling <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+## Prerequisites
+
+[!INCLUDE[](~/includes/7.0-SDK.md)]
+
+## Namespace
+
+The JS interop API described in this article is controlled by attributes in the <xref:System.Runtime.InteropServices.JavaScript?displayProperty=fullName> namespace.
+
+## Enable unsafe blocks
+
+Enable the <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks> property in app's project file, which permits the code generator in the Roslyn compiler to use pointers for JS interop:
+
+```xml
+<PropertyGroup>
+  <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+</PropertyGroup>
+```
+
+> [!WARNING]
+> The JS interop API requires enabling <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks>. Be careful when implementing your own unsafe code in .NET apps, which can introduce security and stability risks. For more information, see [Unsafe code, pointer types, and function pointers](/dotnet/csharp/language-reference/unsafe-code).
+
+In apps generated from the `wasmbrowser` or `wasmconsole` templates, the <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks> property is set in the project file (`.csproj`).
+
+## .NET JavaScript interop on :::no-loc text="WASM":::
+
+The [JS ES6 module](xref:blazor/js-interop/index#javascript-isolation-in-javascript-modules) in `main.js` in a project created from the `wasmbrowser` project template demonstrates JS interop. The relevant APIs are imported from `dotnet.js`. These APIs enable you to set up named modules that can be imported into your C# code and call into methods exposed by your .NET code, including `Program.Main`.
+
+Function calls in the following example:
+
+* `dotnet.withDiagnosticTracing(false).withApplicationArgumentsFromQuery().create()` sets up the .NET WebAssembly runtime without diagnostic tracing and with query string arguments.
```

Comment:
We could probably simplify this to just `dotnet.create()` and then document the other DotnetHostBuilder APIs separately. See the full API definiton here: https://github.com/dotnet/runtime/blob/main/src/mono/wasm/runtime/dotnet.d.ts.